bundles/org.openhab.binding.upnpcontrol/src/main/java/org/openhab/binding/upnpcontrol/internal/handler/UpnpRendererHandler.java

```diff
@@ -411,255 +554,812 @@ protected void setMute(String channel, OnOffType mute) {
-                        newState = PlayPauseType.PAUSE;
-                    }
-                    updateState(channelUID, newState);
-                    break;
+            playingNotification = false;
+            notificationVolume = null;
+            notificationUri = "";
+
+            if (playing) {
+                int pos = posAtNotificationStart;
+                seek(String.format("%02d:%02d:%02d", pos / 3600, (pos % 3600) / 60, pos % 60));
+                play();
             }
-            return;
-        } else {
-            switch (channelUID.getId()) {
-                case VOLUME:
-                    setVolume(getCurrentChannel(), (PercentType) command);
-                    break;
-                case MUTE:
-                    setMute(getCurrentChannel(), (OnOffType) command);
-                    break;
-                case STOP:
-                    if (command == OnOffType.ON) {
-                        updateState(CONTROL, PlayPauseType.PAUSE);
-                        playerStopped = true;
-                        stop();
-                        updateState(TRACK_POSITION, new QuantityType<>(0, SmartHomeUnits.SECOND));
-                    }
-                    break;
-                case CONTROL:
-                    playerStopped = false;
-                    if (command instanceof PlayPauseType) {
-                        if (command == PlayPauseType.PLAY) {
-                            play();
-                        } else if (command == PlayPauseType.PAUSE) {
-                            pause();
-                        }
-                    } else if (command instanceof NextPreviousType) {
-                        if (command == NextPreviousType.NEXT) {
-                            playerStopped = true;
-                            serveNext();
-                        } else if (command == NextPreviousType.PREVIOUS) {
-                            playerStopped = true;
-                            servePrevious();
-                        }
-                    } else if (command instanceof RewindFastforwardType) {
-                    }
-                    break;
+            posAtNotificationStart = 0;
+        }
+    }
+
+    private void playFavorite() {
+        UpnpFavorite favorite = new UpnpFavorite(favoriteName, bindingConfig.path);
+        String uri = favorite.getUri();
+        UpnpEntry entry = favorite.getUpnpEntry();
+        if (!uri.isEmpty()) {
+            String metadata = "";
+            if (entry != null) {
+                metadata = UpnpXMLParser.compileMetadataString(entry);
             }
+            setCurrentURI(uri, metadata);
+            play();
+        }
+    }
 
-            return;
+    void updateFavoritesList() {
+        synchronized (favoriteCommandOptionList) {
+            favoriteCommandOptionList = UpnpControlUtil.favorites(bindingConfig.path).stream()
+                    .map(p -> (new CommandOption(p, p))).collect(Collectors.toList());
+        }
```

Comment:
What is the point of synchronizing this? Also you are changing the instance that `synchronized ` is using for locking which kinda defeats the intended purpose of `synchronized `. What are you trying to make threadsafe? Field assignment is already an atomic operation.